R/FilteredData.R

```diff
@@ -908,6 +866,64 @@ FilteredData <- R6::R6Class( # nolint
 
   ## __Private Methods ====
   private = list(
+    # Handles changes in `active_datanames`.
+    # @param active_datanames (`reactive(1)`) the observed reactive
+    # @param id (`character(1)`) the id of the filter panel
+    # @return a `shiny` observer
+    active_datanames_observer = function(active_datanames, id) {
+      # rather than regenerating the UI dynamically for the dataset filtering,
+      # we instead choose to hide/show the elements
+      # the filters for this dataset are just hidden from the UI, but still applied
+      # optimization: we set `priority = 1` to execute it before the other
+      # observers (default priority 0), so that they are not computed if they are hidden anyways
+      observeEvent(active_datanames(),
+        priority = 1,
+        {
+          logger::log_trace(
+            "FilteredData$srv_filter_panel@1 active datanames: { paste(active_datanames(), collapse = \" \") }"
+          )
+
+          private$hide_inactive_datasets(active_datanames)
+          if (length(active_datanames()) == 0 || is.null(active_datanames())) {
+            # The filter panel emits an event when there are no active datasets
+            # so the parent modules can hide the filter panel if needed.
+            script <- paste0(
+              "const noDatasetsEvent = new Event('noDatasetsEvent');",
+              "document.getElementById('%s').dispatchEvent(noDatasetsEvent);"
+            )
```

Comment:
This is bad as it tries to call a js which is not in this package but in teal. We should find more encapsulated solution instead of making links between html elements in two packages. teal can depend on the js functions from teal.slice but not otherwise. When I read the code I'm concluding that we should first think how to enable $filter_panel_srv and $filter_panel_ui to interact with the `active_datanames` from teal.